runtime/vam/expr/index.go

```diff
@@ -47,7 +48,8 @@ func indexArrayOrSet(sctx *super.Context, vec, indexVec vector.Any) vector.Any {
 		vec, index = view.Any, view.Index
 	}
 	offsets, vals, nulls := arrayOrSetContents(vec)
-	indexVec = promoteToSigned(indexVec)
+	// XXX With uints there's a possiblity of overflow errors getting returned.
```

Comment:
Let's deal with this and add a test for it. I think we just need something like this.

```go
	indexVec = cast.To(sctx, indexVec, super.TypeInt64)
	dyn, _ := indexVec.(*vector.Dynamic)
	var errs []uint32
	var viewIndexes []uint32
	for i := range indexVec.Len() {
		idx := i
		if index != nil {
			idx = index[i]
		}
		if dyn != nil && dyn.TypeOf(uint32(i)).ID() != super.IDInt64 {
			errs = append(errs, i)
			continue
		}
```

Reply:
this doesn't work

Reply:
Alright I pushed a fix for this.